table/block_based/block_based_table_builder.cc

```diff
@@ -1644,44 +1644,40 @@ void BlockBasedTableBuilder::EnterUnbuffered() {
   const size_t kSampleBytes = r->compression_opts.zstd_max_train_bytes > 0
                                   ? r->compression_opts.zstd_max_train_bytes
                                   : r->compression_opts.max_dict_bytes;
+  const size_t kNumBlocksBuffered = r->data_block_and_keys_buffers.size();
+
+  // Abstract algebra teaches us that a finite cyclic group (such as the
+  // additive group of integers modulo N) can be generated by a number that is
+  // coprime with N. Since N is variable (number of buffered data blocks), we
+  // must then pick a prime number in order to guarantee coprimeness with any N.
+  // Picked a Mersenne prime because it's easy on the eyes. No idea about its
+  // properties.
+  //
+  // One downside of this approach is the spread will be poor when
+  // `kPrimeGeneratorRemainder` is small enough that we pick all samples before
+  // cycling through the whole array.
+  const uint64_t kPrimeGenerator = (1ull << 61) - 1;
```

Comment:
Is this the worst choice of prime for some reason? For almost all `kNumBlocksBuffered` the test cases use (e.g., 6, 9, 31, 32, 33) I am seeing `kPrimeGeneratorRemainder == 1` or `kPrimeGeneratorRemainder == kNumBlocksBuffered - 1`. I will change this to something unrelated to powers of two...